cmd/entrypoint/irtype.go

```diff
@@ -0,0 +1,177 @@
+package entrypoint
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type IRResource struct {
+	Active       bool     `json:"_active"`
+	CacheKey     string   `json:"_cache_key,omitempty"`
+	Errored      bool     `json:"_errored"`
+	ReferencedBy []string `json:"_referenced_by,omitempty"`
+	RKey         string   `json:"_rkey,omitempty"`
+	Location     string   `json:"location,omitempty"`
+	Kind         string   `json:"kind"`
+	Name         string   `json:"name"`
+	Namespace    string   `json:"namespace,omitempty"`
+}
+
+type IRClusterHealthCheck struct {
+	IRResource
+}
+
+type IRClusterTarget struct {
+	IP         string `json:"ip"`
+	Port       int    `json:"port"`
+	TargetKind string `json:"target_kind"`
+}
+
+type IRCluster struct {
+	IRResource
+	BarHostname      string               `json:"_hostname"`  // Why this _and_ hostname?
+	BarNamespace     string               `json:"_namespace"` // Why this _and_ namespace?
```

Comment:
I don't like this but I get why it's here for now

Reply:
You and me both. I'll figure out what up with this soon.